java/daikon/dcomp/CalcStack24.java

```diff
@@ -0,0 +1,1047 @@
+import java.lang.classfile.constantpool.IntegerEntry;
+import java.lang.classfile.constantpool.LoadableConstantEntry;
+import java.lang.classfile.constantpool.LongEntry;
+import java.lang.classfile.constantpool.MethodHandleEntry;
+import java.lang.classfile.constantpool.MethodTypeEntry;
+import java.lang.classfile.constantpool.StringEntry;
+import java.lang.classfile.instruction.BranchInstruction;
+import java.lang.classfile.instruction.ConstantInstruction;
+import java.lang.classfile.instruction.DiscontinuedInstruction;
+import java.lang.classfile.instruction.ExceptionCatch;
+import java.lang.classfile.instruction.FieldInstruction;
+import java.lang.classfile.instruction.InvokeDynamicInstruction;
+import java.lang.classfile.instruction.InvokeInstruction;
+import java.lang.classfile.instruction.LineNumber;
+import java.lang.classfile.instruction.LoadInstruction;
+import java.lang.classfile.instruction.LookupSwitchInstruction;
+import java.lang.classfile.instruction.NewMultiArrayInstruction;
+import java.lang.classfile.instruction.NewObjectInstruction;
+import java.lang.classfile.instruction.NewPrimitiveArrayInstruction;
+import java.lang.classfile.instruction.NewReferenceArrayInstruction;
+import java.lang.classfile.instruction.StoreInstruction;
+import java.lang.classfile.instruction.SwitchCase;
+import java.lang.classfile.instruction.TableSwitchInstruction;
+import java.lang.classfile.instruction.TypeCheckInstruction;
+import java.lang.constant.ClassDesc;
+import java.lang.constant.MethodTypeDesc;
+import java.util.List;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/** This class calculates the state of the operand stack via simulation. */
+public final class CalcStack24 {
+
+  /** Don't allow others to create a new CalcStack24 object. */
+  private CalcStack24() {}
+
+  /** ClassDesc for 'null'. */
+  static final ClassDesc nullCD = ClassDesc.of("fake.ClassDecs.for.null");
+
+  /**
+   * Calculates changes in contents of operand stack based on the symbolic execution of a Java
+   * bytecode instruction. Note that we assume the class file is valid and make no attempt to verify
+   * the code's correctness.
+   *
+   * @param mgen method containing the instruction (currently unused)
+   * @param minfo for the given method's code (currently unused)
+   * @param ce instruction to be interpreted
+   * @param inst_index index of ce in code element list
+   * @param stack current state of operand stack
+   * @return true if should continue with next instruction, false otherwise
+   */
+  @SuppressWarnings("fallthrough")
+  static boolean calcOperandStack(
+      MethodGen24 mgen,
+      MethodGen24.MInfo24 minfo,
+      CodeElement ce,
+      int inst_index,
+      OperandStack24 stack) {
+
+    if (DCInstrument24.debugOperandStack) {
+      System.out.println("code element: " + ce);
+    }
+    switch (ce) {
+      case Instruction inst -> {
+        if (DCInstrument24.stacks[inst_index] != null) {
```

Comment:
To make the overall structure of this method clearer (that is, to show what happens for cases other than `case Instruction inst -> {`), I suggest abstracting out the body of `case Instruction inst -> {` into a method that takes an `Instruction`.